rust/parser/typeql.pest

```diff
@@ -114,73 +114,104 @@ type_constraint = { ABSTRACT
-                 | VAR_ ~ IID ~ IID_ ~ ( "," ~ attributes )?
-                 | VAR_ ~ attributes
+variable_thing = { VAR_CONCEPT_ ~ ISA_ ~ type_ ~ ( "," ~ attributes )?
+                 | VAR_CONCEPT_ ~ IID ~ IID_ ~ ( "," ~ attributes )?
+                 | VAR_CONCEPT_ ~ attributes
                  }
 
-variable_relation = { VAR_? ~ relation ~ ISA_ ~ type_ ~ ( "," ~ attributes )?
-                    | VAR_? ~ relation ~ attributes? }
+variable_relation = { VAR_CONCEPT_? ~ relation ~ ISA_ ~ type_ ~ ( "," ~ attributes )?
+                    | VAR_CONCEPT_? ~ relation ~ attributes? }
 
-variable_attribute = { VAR_? ~ predicate ~ ISA_ ~ type_ ~ ( "," ~ attributes )?
-                     | VAR_? ~ predicate ~ attributes? }
+variable_attribute = { VAR_CONCEPT_? ~ predicate ~ ISA_ ~ type_ ~ ( "," ~ attributes )?
+                     | VAR_CONCEPT_? ~ predicate ~ attributes? }
 
 // RELATION CONSTRUCT ==========================================================
 
 relation = { "(" ~ role_player ~ ( "," ~ role_player )* ~ ")" }                 // A list of role players in a Relations
 role_player = { type_ ~ ":" ~ player | player }                                 // The Role type and player variable or just the player variable
-player = { VAR_ }                                                               // A player is just a variable
+player = { VAR_CONCEPT_ }                                                       // A player is just a variable
 
 // ATTRIBUTE CONSTRUCT =========================================================
 
 attributes = { attribute ~ ( "," ~ attribute )* }
-attribute = { HAS ~ label ~ ( VAR_ | predicate ) | HAS ~ VAR_ }                 // ownership by labeled variable or value or just value
+attribute = { HAS ~ label ~ ( VAR_ | predicate )
+            | HAS ~ VAR_CONCEPT_
+            }                                                                   // ownership by labeled variable or value or just value
 
-// ATTRIBUTE VALUATION CONSTRUCTS ==============================================
+// PREDICATE CONSTRUCTS ========================================================
 
 predicate = { value | predicate_equality ~ predicate_value | predicate_substring ~ STRING_ }
 
-predicate_equality = { EQ | NEQ | GTE | GT | LTE | LT }
+predicate_equality = { EQ | NEQ | GTE | GT | LTE | LT
+                     | ASSIGN                                                   // Backwards compatibility till 3.0
+                     }
+
 predicate_substring = { CONTAINS | LIKE }
 
 predicate_value = { value | VAR_ }
 
-// SCHEMA CONSTRUCT =============================================================
+// EXPRESSION CONSTRUCTS =======================================================
+
+expression = { expression_base ~ ( infix ~ expression_base )* }
+
+expression_base = _{ VAR_
+                   | expression_function | value
+                   | paren_expr
+                   }
+
+infix = _{ POWER | MULTIPLY | DIVIDE | MODULO | ADD | SUBTRACT }
```

Comment:
- `arithmetic_operator`, because we have this name in `token.rs`.
- `infix_operator` or `binary_operator`, because if we add unary operator(s), they can't be united with binary in this case. `binary` is consistent with existent `Arity` enum, prepared for future usage.

Reply:
@flyingsilverfin : "EXPONENTIATE"?

Reply:
Discussed in person